app/controllers/todos_controller.rb

```diff
@@ -0,0 +1,56 @@
+class TodosController < ApplicationController
+  def index
+    @incomplete_todos = Todo.where(completed: false).order(due_date: :asc)
+    @complete_todos = Todo.where(completed: true).order(due_date: :asc)
+  end
+
+  def show
+    @todo = Todo.find(params[:id])
+  end
+
+  def new
+    @todo = Todo.new
+  end
+
+  def create
+    @todo = Todo.new(todo_params)
+    if @todo.save
+      flash[:notice] = "You are getting baguettes" # todo fix this
+      redirect_to todos_path
+    else
+      render "new"
+    end
+  end
+
+  def edit
+    @todo = Todo.find(params[:id])
+  end
+
+  def update
+    @todo = Todo.find(params[:id])
+    if @todo.update(todo_params)
+      flash[:notice] = "You added jelly to your baguette"
+      redirect_to todos_path
+    else
+      render "edit"
+    end
+  end
+
+  def destroy
+    @todo = Todo.find(params[:id])
+
+    if @todo.destroy
+      flash[:notice] = "You destroyed your sandwich"
+      redirect_to todos_path
+    else
+      flash[:notice] = "The delete failed."
+      redirect_to todos_path
```

Comment:
In what type of situation do you envision a failed delete? Depending on your answer to that question, would you ever want to instead redirect to the particular todo (if it exists) that failed? I had this question while we were building but didn't have time to really dig into it.

Reply:
We had the same question! We figured that we didn't expect it to fail, but at least if there was a failure for whatever reason it would flag it for the user somehow.

Reply:
We almost put that in our code, too. I don't see any reason NOT to include it, even though I can't think of why it would fail either.